src/MyEventEmitter.ts

```diff
@@ -0,0 +1,150 @@
+'use strict';
+
+class Listener {
+  eventName;
+  callback;
+  once;
+
+  constructor(
+    eventName,
+    callback,
+    once = false,
+  ) {
+    this.eventName = eventName;
+    this.callback = callback;
+    this.once = once;
+  }
+}
+
+export class MyEventEmitter {
+  listenersMap = new Map();
+
+  on(eventName, callback) {
+    this.addListener({ eventName, callback });
+
+    return this;
+  }
+
+  once(eventName, callback) {
+    this.addListener({
+      eventName,
+      callback,
+      once: true,
+    });
+
+    return this;
+  }
+
+  prependListener(eventName, callback) {
+    this.addListener({
+      eventName,
+      callback,
+      prepend: true,
+    });
+
+    return this;
+  }
+
+  prependOnceListener(eventName, callback) {
+    this.addListener({
+      eventName,
+      callback,
+      prepend: true,
+      once: true,
+    });
+
+    return this;
+  }
+
+  addListener({
+    eventName,
+    callback,
+    prepend = false,
+    once = false,
+  }) {
+    const newListener = new Listener(eventName, callback, once);
+
+    const eventListeners = this.listenersMap.get(eventName);
+
+    if (eventListeners) {
+      if (prepend) {
+        eventListeners.unshift(newListener);
+      } else {
+        eventListeners.push(newListener);
+      }
+    } else {
+      this.listenersMap.set(eventName, [newListener]);
+    }
```

Comment:
There is some technic to make this (repeated often if you're using Maps) piece of code.
```suggestion
    let eventListeners = this.listenersMap.get(eventName);
    
    if (!eventListeners) {
      eventListeners = [];
      this.listenersMap.set(eventName, eventListeners);
    }

    if (prepend) {
      eventListeners.unshift(newListener);
    } else {
      eventListeners.push(newListener);
    }
```

Reply:
Yeah, looks better - TU